lib/topological_inventory/persister/worker.rb

```diff
@@ -10,11 +10,12 @@ module Persister
     class Worker
       include Logging
 
-      def initialize(messaging_client_opts = {})
+      def initialize(opts = {})
+        messaging_client_opts = opts.select { |k, _| %i[host port].include?(k) }
         self.messaging_client_opts = default_messaging_opts.merge(messaging_client_opts)
 
         InventoryRefresh.logger = logger
-        self.metrics = TopologicalInventory::Persister::Metrics.new
+        self.metrics = TopologicalInventory::Persister::Metrics.new(opts[:metrics_port].to_i)
```

Comment:
Can this `to_i` live in the bin file rather than here?

Reply:
I tried, but if I do it, then all specs will be broken because default value is nil, not 9394.
So it's possible, but with many specs changes

Reply:
It's only two lines ... Here's the patch:

```patch
diff --git a/bin/topological_inventory-persister b/bin/topological_inventory-persister
index 8557ccf..613c6e0 100755
--- a/bin/topological_inventory-persister
+++ b/bin/topological_inventory-persister
@@ -8,7 +8,7 @@ require "topological_inventory/core/ar_helper"
 opts = {
   :host => ENV["QUEUE_HOST"] || "localhost",
   :port => ENV["QUEUE_PORT"] || 9092,
-  :metrics_port => ENV["METRICS_PORT"] || 9394 # 0 disables metrics
+  :metrics_port => (ENV["METRICS_PORT"] || 9394).to_i # 0 disables metrics
 }
 
 TopologicalInventory::Core::ArHelper.database_yaml_path = Pathname.new(__dir__).join("../config/database.yml")
diff --git a/lib/topological_inventory/persister/worker.rb b/lib/topological_inventory/persister/worker.rb
index 29cc229..4d8d848 100644
--- a/lib/topological_inventory/persister/worker.rb
+++ b/lib/topological_inventory/persister/worker.rb
@@ -15,7 +15,7 @@ module TopologicalInventory
         self.messaging_client_opts = default_messaging_opts.merge(messaging_client_opts)
 
         InventoryRefresh.logger = logger
-        self.metrics = TopologicalInventory::Persister::Metrics.new(opts[:metrics_port].to_i)
+        self.metrics = TopologicalInventory::Persister::Metrics.new(opts[:metrics_port])
       end
 
       def run
diff --git a/spec/persister/refresh_worker_helper.rb b/spec/persister/refresh_worker_helper.rb
index 5277272..5dbf33f 100644
--- a/spec/persister/refresh_worker_helper.rb
+++ b/spec/persister/refresh_worker_helper.rb
@@ -6,7 +6,7 @@ module TestInventory
 
       allow(client).to receive(:subscribe_topic).and_yield(message)
 
-      described_class.new.run
+      described_class.new(:metrics_port => 9394).run
       source.reload
     end
   end
diff --git a/spec/persister/worker_spec.rb b/spec/persister/worker_spec.rb
index 178229c..ea0c3f5 100644
--- a/spec/persister/worker_spec.rb
+++ b/spec/persister/worker_spec.rb
@@ -26,7 +26,7 @@ describe TopologicalInventory::Persister::Worker do
       allow(client).to receive(:publish_message)
       allow(client).to receive(:subscribe_topic).and_yield(message)
 
-      described_class.new.run
+      described_class.new(:metrics_port => 9394).run
       source.reload
     end
 
```

Reply:
@carbonin I don't think why it's better if you have to pass method argument every time and this argument is often useless - as metrics are optional, then argument should be optional too.
But it's not a big thing, done.

Reply:
> this argument is often useless

This is not true. This argument is always used in production.

> then argument should be optional too

That's a good point. If we're always passing the value then maybe it shouldn't be in an args hash and should be a proper parameter. That's how I'm doing it in the rest of the components (e.g. https://github.com/ManageIQ/topological_inventory-host_inventory_sync/pull/8/files#diff-9b22de196e3f34380291cc423dcc0094R16)

But if you don't want to change it again, that's fine too.